ytypes/unmarshal.go

```diff
@@ -29,6 +30,49 @@ type UnmarshalOpt interface {
 	IsUnmarshalOpt()
 }
 
+// ComplianceErrors contains the compliance errors encountered from an Unmarshal operation.
+type ComplianceErrors struct {
+	// Errors represent generic errors for now, until we make a decision on what specific types
+	// of errors should be returned.
+	Errors []error
+}
+
+func (c *ComplianceErrors) Error() string {
+	if c == nil {
+		return ""
+	}
+
+	var b strings.Builder
+	b.WriteString("Noncompliance errors:")
+	if len(c.Errors) != 0 {
+		for _, e := range c.Errors {
+			b.WriteString("\n\t")
+			b.WriteString(e.Error())
+		}
+	} else {
+		b.WriteString(" None")
+	}
+	b.WriteString("\n")
+	return b.String()
+}
```

Comment:
Can you make it so that we have coverage for this? A good way is to use this: https://pkg.go.dev/github.com/openconfig/gnmi/errdiff#Substring and just check for the presence of "Noncompliance errors" whenever `hasBestEffortUnmarshal` returns true. I think it makes sense to just add this option for one other random passing test to get coverage for the nil case. Thanks!

Reply:
Is it okay to have the coverage in gnmi_test.go? Currently, the BestEffortUnmarshal only has an effect when used on the UnmarshalSetRequest function. I could also add functionality for this to the unmarshalGeneric function if you think that'd be a good idea.

Reply:
It's ok that `ytypes.Unmarshal` doesn't support this option. Actually it would be good to add logic there to error out if this option is provided saying it's not supported. It's somewhat involved to get it working there since it deals with JSON and there are a lot of places where errors can happen so I'm ok with not supporting it.

As for coverage of the `String()` method I'm fine with either having direct coverage in unmarshal_test.go or via Substring.

Oh yes, since this also affects UnmarshalNotifications can you add a test case for that as well? Thanks.

Reply:
Added!